File: src/main/java/seremis/geninfusion/soul/gene/newAI/mate/GeneAIMate.java
package seremis.geninfusion.soul.gene.newAI.mate;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.SoulHelper;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAIMate extends MasterGene {

    public GeneAIMate() {
        addControlledGene(SoulHelper.geneRegistry.getGene(Genes.GENE_AI_MATE_MOVE_SPEED));
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/scala/seremis/geninfusion/soul/gene/newAI/GeneAIMate.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAIMate extends MasterGene(classOf[Boolean]) {
    addControlledGene(Genes.GeneAIMateIndex)
    addControlledGene(Genes.GeneAIMateMoveSpeed)
}
